### DISCUSSION TITLE
support query prompt via prompt id

### DISCUSSION THREAD

**Discussion by hu9029** [langfuse/langfuse#2184] [Ideas]
### Describe the feature or potential improvement

Scenario: Testing the scores returned by different versions of a prompt

Background: In this testing scenario, a trace contains multiple Observations. To correlate Observations from two different traces, the prompts need to be associated. However, only the prompt_id is available, and there might be cases where different prompt_ids have the same prompt_name. Therefore, it is necessary to find the corresponding prompt_name through the prompt_id to locate the associated Observations.

Problem: Currently, only prompt_id is returned. How can we find the corresponding prompt_name using prompt_id to correctly associate the relevant Observations?

Suggestion: Add an API to retrieve the prompt_name using the prompt_id.

**Comment by marcklingen** [comment]
thanks for raising this. Langfuse should return the name and version on the generation API as well to make it easy to fetch the related prompt. Creating an issue for this

**Reply by marcklingen** [reply]
tracking this here: https://github.com/langfuse/langfuse/issues/2341#issue-2352148119

**Reply by Lucien0907** [reply]
Sorry I was having a similar issue, I had prompt id as meta data associated with one of my experiment run but no prompt name or version, **trying to find a api to retrieve name and version from id but no success**, **there still don't seem be a dedicated API for doing this?** i am able to get prompt name and version directly from observation id tho using this: /api/public/observations/{observationId}​
